util/c_function_graph.py

```
@@ -95,7 +111,77 @@ def get_functions_defined_in_file(file_path: Path) -> list[CFunction]:
+        for path in paths:
+            for fn in get_functions_from_file(path):
+                if fn.name not in self.functions:
+                    fn.is_external_function = True
+                    self.functions[fn.name] = fn
+                    self.call_graph.add_node(fn)
+
+        # Re-wire edges now that the full function set is known.
+        for func in self.functions.values():
+            for callee_name in func.callee_names:
+                if callee := self.functions.get(callee_name):
+                    if not self.call_graph.has_edge(func, callee):
```

Ruff (0.15.9)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)